metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery.py

```diff
@@ -1096,106 +1117,74 @@ def get_tables_for_dataset(
+                        # When table is only a shard we use dataset_name as table_name
+                        sharded_tables[table_name] = table
+                        continue
                     else:
-                        table_count = table_count + 1
-                        table_items[table.table_id] = table
-
-                    if str(table_identifier).startswith(
-                        self.config.temp_table_dataset_prefix
-                    ):
-                        logger.debug(
-                            f"Dropping temporary table {table_identifier.table}"
+                        stored_table_identifier = BigqueryTableIdentifier(
+                            project_id=project_id,
+                            dataset=dataset_name,
+                            table=sharded_tables[table_name].table_id,
                         )
-                        self.report.report_dropped(table_identifier.raw_table_name())
-                        continue
-
-                    if (
-                        table_count % self.config.number_of_datasets_process_in_batch
-                        == 0
-                    ):
-                        bigquery_tables.extend(
-                            BigQueryDataDictionary.get_tables_for_dataset(
-                                conn,
-                                project_id,
-                                dataset_name,
-                                table_items,
-                                with_data_read_permission=self.config.profiling.enabled,
-                            )
+                        (
+                            _,
+                            stored_shard,
+                        ) = BigqueryTableIdentifier.get_table_and_shard(
+                            stored_table_identifier.raw_table_name()
                         )
-                        table_items.clear()
-
-                # Sharded tables don't have partition keys, so it is safe to add to the list as
-                # it should not affect the number of tables will be touched in the partitions system view.
-                # Because we have the batched query of get_tables_for_dataset to makes sure
-                # we won't hit too many tables queried with partitions system view.
-                # The key in the map is the actual underlying table name and not the friendly name and
-                # that's why we need to get the actual table names and not the normalized ones.
-                table_items.update(
-                    {value.table_id: value for value in sharded_tables.values()}
-                )
-
-                if table_items:
+                        # When table is none, we use dataset_name as table_name
+                        table_name = table_identifier.get_table_name().split(".")[-1]
+                        assert stored_shard
+                        if stored_shard < shard:
```

Comment:
This condition makes very little sense to me as an unfamiliar reader. Consider adding a comment to explain